eth-connector/src/lib.rs

```diff
@@ -304,36 +225,45 @@ impl EthConnectorContract {
         balances
     }
 
-    #[must_use]
-    pub fn get_bridge_prover(&self) -> AccountId {
-        self.connector.prover_account.clone()
-    }
-
-    #[access_control_any(roles(Role::DAO))]
-    pub fn set_prover(&mut self, account_id: AccountId) {
-        self.connector.prover_account = account_id;
-    }
-
-    #[access_control_any(roles(Role::DAO))]
-    pub fn set_min_proof_acceptance_height(&mut self, height: u64) {
-        self.connector.min_proof_acceptance_height = height;
-    }
-
     #[payable]
     #[access_control_any(roles(Role::DAO))]
     pub fn set_aurora_engine_account_id(&mut self, new_aurora_engine_account_id: AccountId) {
         assert_one_yocto();
-        self.connector.aurora_engine_account_id = new_aurora_engine_account_id;
+        self.aurora_engine_account_id = new_aurora_engine_account_id;
     }
 
     #[must_use]
     pub fn get_aurora_engine_account_id(&self) -> AccountId {
-        self.connector.aurora_engine_account_id.clone()
+        self.aurora_engine_account_id.clone()
     }
 
     #[must_use]
-    pub fn get_config(&self) -> EthConnector {
-        self.connector.clone()
+    pub fn get_controller(&self) -> AccountId {
+        self.controller.clone()
+    }
+
+    #[payable]
+    pub fn mint(
+        &mut self,
+        account_id: AccountId,
+        amount: U128,
+        msg: Option<String>,
+    ) -> PromiseOrValue<U128> {
+        self.assert_controller();
+        self.register_if_not_exists(&account_id);
+        if let Some(msg) = msg {
+            self.mint_eth_on_near(&env::predecessor_account_id(), amount.into());
```

Comment:
Shouldn't we mint on current account instead of `env::predecessor_account_id()`?